File: framework/languages/java/e2e/ResilienceLifecycle/Client/src/main/java/systems/zlink/e2e/resiliencelifecycle/client/Scenarios/RlD3DispatchErrorEvidenceScenario.java
package systems.zlink.e2e.resiliencelifecycle.client.Scenarios;

import systems.zlink.e2e.resiliencelifecycle.client.Support.ConsumerScenarioClient;

public final class RlD3DispatchErrorEvidenceScenario {
    private RlD3DispatchErrorEvidenceScenario() {
    }

    public static void run(ConsumerScenarioClient consumer) {
        consumer.runMode("rl-d3");
    }
}

File: framework/languages/java/e2e-kotlin/ResilienceLifecycle/Client/src/main/kotlin/systems/zlink/e2e/kotlin/resiliencelifecycle/Scenarios/RlD3DispatchErrorEvidenceScenario.kt
package systems.zlink.e2e.kotlin.resiliencelifecycle

import java.time.Duration

fun ClientScenarioContext.runDispatchErrorEvidenceScenario() {
    try {
        client.requestToChannel(Contracts.CHANNEL, Contracts.UnhandledRequest("d3-missing-handler"))
            .timeout(Duration.ofSeconds(3))
            .await(Contracts.WorkReply::class.java)
        throw IllegalStateException("RL-D3 missing handler request unexpectedly completed")
    } catch (_: RuntimeException) {
        waitForDispatchErrorAny("UnhandledRequest", adminA(), adminB())
    }
    val followUp = client.requestToChannel(Contracts.CHANNEL, Contracts.WorkRequest("d3-follow-up"))
        .timeout(Duration.ofSeconds(3))
        .await(Contracts.WorkReply::class.java)
    ensure(followUp.value() == "work:d3-follow-up", "RL-D3 follow-up payload mismatch")
    println("scenario RL-D3 passed")
}
